internal/command/meta_backend.go

```diff
@@ -320,6 +320,9 @@ func (m *Meta) BackendForPlan(settings plans.Backend) (backend.Enhanced, tfdiags
 
 	configureDiags := b.Configure(newVal)
 	diags = diags.Append(configureDiags)
+	if configureDiags.HasErrors() {
```

Comment:
would write this check under the `configureDiags` assignment before appending. Besides that awesome work... I know a small change like this must have been painful to catch! 🔥